pyroomacoustics/directivities.py

```diff
@@ -0,0 +1,183 @@
+import numpy as np
+from pyroomacoustics.utilities import Options
+
+
+class DirectivityPattern(Options):
+    FIGURE_EIGHT = "figure_eight"
+    HYPERCARDIOID = "hypercardioid"
+    CARDIOID = "cardioid"
+    SUBCARDIOID = "subcardioid"
+    OMNI = "omni"
+
+
+PATTERN_TO_CONSTANT = {
+    DirectivityPattern.FIGURE_EIGHT: 0,
+    DirectivityPattern.HYPERCARDIOID: 0.25,
+    DirectivityPattern.CARDIOID: 0.5,
+    DirectivityPattern.SUBCARDIOID: 0.75,
+    DirectivityPattern.OMNI: 1.0
+}
+
+
+class DirectionVector(object):
+    """
+    Parameters
+    ----------
+    azimuth : float
+    colatitude : float, optional
+        Default to PI / 2, only XY plane.
+    degrees : bool
+        Whether provided values are in degrees (True) or radians (False).
+    """
+    def __init__(self, azimuth, colatitude=None, degrees=True):
+        if degrees is True:
+            azimuth = np.radians(azimuth)
+            if colatitude is not None:
+                colatitude = np.radians(colatitude)
+        self._azimuth = azimuth
+        if colatitude is None:
+            colatitude = np.pi / 2
+        assert colatitude <= np.pi and colatitude >= 0
+        self._colatitude = colatitude
+
+    def get_azimuth(self, degrees=False):
+        if degrees:
+            return np.degrees(self._azimuth)
+        else:
+            return self._azimuth
+
+    def get_colatitude(self, degrees=False):
+        if degrees:
+            return np.degrees(self._colatitude)
+        else:
+            return self._colatitude
+
+
+class Directivity(object):
+    """
+
+    Parameters
+    ----------
+    orientation : DirectionVector
+        Indicates direction of the pattern.
+    pattern : str
+        One of directivities in `DirectivityPattern`.
+        TODO : support arbitrary directivities.
+
+    """
+    def __init__(self, orientation, pattern):
+        assert pattern in DirectivityPattern.values()
+        assert isinstance(orientation, DirectionVector)
+        self._orientation = orientation
+        self._pattern = pattern
+        self._p = PATTERN_TO_CONSTANT[pattern]
+
+    def get_directivity_pattern(self):
```

Comment:
Here we could use the `__repr__` or `__str__` special methods.

Also, recently I quite like the property decorator for getter and setter. Here it would be like this:
```python
@property
def directivity_pattern(self):
    return self._pattern
```